api/v2/serializers/post/token_update.py

```diff
@@ -23,30 +24,61 @@ def validate(self, data):
         IF identity is found validation will:
         - Ensure that user/token produces a valid driver
         """
-        validated_data = data
-        self.validate_token_with_driver(data['provider'], data['username'], data['project_name'], data['token'])
-        return validated_data
+        logger.info("TokenUpdateSerializer:validate called")
+        # create will fail if it is an invalid token, but we need to create an identity first
```

Comment:
The identity is not a relation that maps a specific user to a provider.  It does not confer that the user is logged in to the provider.  That is currently done when the token is stored as a user credential.

The self.validate_token_with_driver function puts the cart before the horse.  We have just obtained the token from keystone so why spend the extra time confirming that the keystone token that we just got from keystone is able to be validated by keystone?